src/pages/Admin/Venue/Details/VenueDetails.tsx

```diff
@@ -0,0 +1,254 @@
+          </Link>
+          <Button>Preview landing page</Button>
+
+          <S.AdminList>
+            <S.AdminListTitle>Party admins</S.AdminListTitle>
+
+            {ownersData.length > 0 &&
+              ownersData.map((owner: Owner) => (
+                <S.AdminItem key={owner.id}>
+                  <S.AdminPicture backgroundImage={owner.pictureUrl} />
+                  <S.AdminItemTitle>{owner.partyName}</S.AdminItemTitle>
+                </S.AdminItem>
+              ))}
+
+            <Button onClick={() => setShowOwnersModal(true)}>
+              Invite an admin
+            </Button>
+          </S.AdminList>
+        </S.HeaderActions>
+      </S.Header>
+
+      <S.Main>
+        <MapPreview
+          venueId={venueId!}
+          venueName={name}
+          mapBackground={mapBackgroundImageUrl}
+          rooms={rooms}
+        />
+
+        {!!mapBackgroundImageUrl && (
+          <>
+            <S.RoomActions>
+              <S.RoomCounter>{rooms ? rooms.length : "0"} Rooms</S.RoomCounter>
+              <Button onClick={() => toggleRoomModal()} gradient>
+                Add a room
+              </Button>
+            </S.RoomActions>
+          </>
+        )}
+
+        {!!rooms && !!mapBackgroundImageUrl && (
+          <S.RoomWrapper>
+            {rooms.map((room: AnyRoom, index: number) => (
+              <RoomCard
+                key={room.title}
+                room={room}
+                venueId={venueId!}
+                editHandler={() => handleEditRoom(room, index)}
+                onEventHandler={handleRoomEvent}
+                roomIndex={index}
+              />
+            ))}
+          </S.RoomWrapper>
+        )}
+      </S.Main>
+
+      <RoomModal
+        isVisible={modalOpen}
+        venueId={venueId!}
+        onSubmitHandler={() => setModalOpen(false)}
+        onClickOutsideHandler={() => setModalOpen(false)}
+      />
+
+      {editingRoom && (
+        <RoomEdit
+          isVisible={!!editingRoom}
+          onClickOutsideHandler={() => setEditingRoom(null)}
+          room={editingRoom}
+          submitHandler={handleEditRoomSave}
+          deleteHandler={() => setShowDeleteModal(true)}
+        />
+      )}
+
+      {editingRoom && (
+        <RoomDeleteModal
+          show={showDeleteModal}
+          onHide={() => setShowDeleteModal(false)}
+          venueId={venueId!}
+          room={editingRoom}
+          onDeleteRedirect={`/admin_v2/venue/${venueId}`}
+          onDelete={() => {
+            setShowDeleteModal(false);
+            setEditingRoom(null);
+          }}
+        />
+      )}
+
+      <VenueOwnersModal
+        visible={showOwnersModal}
+        onHide={() => setShowOwnersModal(false)}
+        venue={venue}
+      />
+
+      <AdminEventModal
+        show={showEventModal}
+        venueId={venueId!}
+        onHide={() => setShowEventModal(false)}
+        setEditedEvent={() => {}}
+        setShowDeleteEventModal={() => {}}
```

Comment:
.

Reply:
Probably `noopFunc` or make these params optional.

I would also consider whether there is a better 'higher level' function we could be passing down instead of the raw `setEditedEvent`, `setShowDeleteEventModal`, etc.